overlord/hookstate/ctlcmd/is_connected.go

```diff
@@ -0,0 +1,101 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/overlord/ifacestate"
+)
+
+type isConnectedCommand struct {
+	baseCommand
+
+	Positional struct {
+		PlugOrSlotSpec string `positional-args:"true" positional-arg-name:":<plug|slot>"`
+	} `positional-args:"yes"`
+}
+
+var shortIsConnectedHelp = i18n.G(`The is-connected returns zero if the given plug or slot is connected, and non-zero otherwise`)
+var longIsConnectedHelp = i18n.G(`
+The is-connected returns zero if the given plug or slot of the calling snap is
+connected, and non-zero otherwise.
+
+$ snapctl is-connected plug
+$ echo $?
+1
+
+Snaps can only query their own plugs and slots - snap name is implicit and implied by the snapctl execution context.
+`)
+
+func init() {
+	addCommand("is-connected", shortIsConnectedHelp, longIsConnectedHelp, func() command {
+		return &isConnectedCommand{}
+	})
+}
+
+func (c *isConnectedCommand) Execute(args []string) error {
+	plugOrSlot := c.Positional.PlugOrSlotSpec
+	if plugOrSlot == "" {
+		return fmt.Errorf(i18n.G("plug or slot name not provided"))
```

Comment:
this check wouldn't be needed if `PlugOrSlotSpec` were tagged as required. Instead the user would get
```the required argument `:<plug|slot>` was not provided```.